lab-james/src/components/search-form/index.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import superagent from 'superagent';
+import PropTypes from 'prop-types';
+import './search-form.scss';
+
+class SearchForm extends React.Component {
+
+  constructor(props){
+    super(props);
+    this.performSearch = this.performSearch.bind(this);
+    this.changeState = this.changeState.bind(this);
+    this.errorState = this.errorState.bind(this);
+    this.state = {
+      searchFormBoard: '',
+      searchFormLimit: 15,
+      error: 'working'
+    }
+  }
+
+  errorState(state){
+    this.setState({error: state});
+  }
+
+  changeState(e){
+    this.setState({[e.target.id]: e.target.value})
+  }
+
+  performSearch(e){
+    e.preventDefault();
+    const redditAPI = `http://www.reddit.com/r/${this.state.searchFormBoard}.json?limit=${this.state.searchFormLimit}`;
+    superagent.get(redditAPI)
+      .then(results => {
+        this.errorState('working')
+        this.props.setStateFromSearch(results.body.data.children);
+      })
+      .catch(err => {
+        this.errorState('error');
+        console.log(err);
```

Comment:
I'd probably move this console log to the errorState function, could trim this down by a line or two.